src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs

```diff
@@ -89,13 +89,23 @@ private IEnumerable<string> ApplyCapitalization(IEnumerable<string> words)
         private string CapitalizeFirstLetter(string word)
```

Comment:
I should observe that I have no idea what this code does in the case of surrogate pairs.

Reply:
Would it make sense to test if the first char is already upper case, and just return the input string; i.e. bypassing `ToCharArray` and `new string(chars)` completely?

Reply:
Why yes, yes it would.

Reply:
What I meant was something like this (avoid allocations if `word` does not need to change).

```csharp
private string CapitalizeFirstLetter(string word) {
  if (word == null || word.Length < 1 || char.IsUpper(word[0]))) return word;
  var chars = word.ToCharArray();
  chars[0] = char.ToUpper(chars[0]);
  return new string(chars);
}
```

Reply:
@nil4 Yep, understood perfectly from the start.